test/units/module_utils/parsing/test_convert_bool.py

```diff
@@ -45,14 +45,14 @@ def test_junk_values_nonstrict(self):
         assert boolean(object(), strict=False) is False
 
     def test_junk_values_strict(self):
-        with pytest.raises(TypeError):
-            assert boolean("flibbity", strict=True) is False
+        with pytest.raises(TypeError, match="The value 'flibbity' is not"):
+            boolean("flibbity", strict=True)
 
-        with pytest.raises(TypeError):
-            assert boolean(42, strict=True) is False
+        with pytest.raises(TypeError, match="The value '42' is not"):
+            boolean(42, strict=True)
 
-        with pytest.raises(TypeError):
-            assert boolean(42.0, strict=True) is False
+        with pytest.raises(TypeError, match="The value '42.0' is not"):
```

Comment:
I recommend making it very clear that these are all regexes:
```suggestion
        with pytest.raises(TypeError, match=r"^The value '42\.0' is not"):
```

Seeing regular strings often tricks people into thinking that this is just substring matching, when it's not.